### ISSUE TITLE
Shared handles

### ISSUE THREAD

**Issue by Agrael1** [KhronosGroup/Vulkan-Hpp#1585]
Hello, I have seen the UniqueHandle approach, 

But while creating a layer library I came to the conclusion, that sometimes you'd really benefit from having shared handles for correct order of destruction.

I made them like a shared_pointer, but with parent, so control block actually refers to parent, in order to ensure correct destruction.

```
shared_handle<Ty>:
/       *[control_block]       \ --------->[shared_handle<parent>, refcount, (allocator, loader)]
\       *[handle]                  /
```
may this be of interest? If so, I could port my handles. Although there would be a great benefit if the control block could be allocated with the handle in one allocation, there is no way to implement that right now.

**Comment by asuessenbach** [comment]
Did you have a look at the classes provided by vulkan_raii.hpp?
Besides other advantages, those classes have complete constructors and destructors, and thus can be used with std::shared_ptr.

**Comment by Agrael1** [comment]
> Did you have a look at the classes provided by vulkan_raii.hpp? Besides other advantages, those classes have complete constructors and destructors, and thus can be used with std::shared_ptr.

Yes, but the raii wrappers don't give any guarantee of correct destruction order. If your device is destroyed before swapchain, then you get an error. Shared pointer don't do the trick either, but rather introduce overhead, allocating pointers on heap.

**Comment by asuessenbach** [comment]
I see... as each child holds a shared_handle to its parent, that parent won't be destroyed before all its children are destroyed.
So, yes, I'd be interested to see your approach. Feel free to provide a PR and we can discuss your approach in detail.

**Comment by asuessenbach** [comment]
Resolved with #1619.
